### PR TITLE
release_gen_payload: fix default_imagestream_base_name usage

### PR DESCRIPTION
**Pull request by sosiouxme**
this method changed signature. scan_sources was using the previous
signature and began silently failing, causing it not to detect that
rhcos has an update.

the usage of rhcos.latest_rhcos_build_id also changed out from under
scan_sources - fixed.

### CODE REVIEW

**Comment by sosiouxme** [comment]
/hold
before I merge this, we should probably understand why this is the output now:
```
rhcos:
-   changed: false
    name: 4.8-x86_64
    reason: latest RHCOS build is still 48.84.202109161436-0 -- no change from istag
-   changed: true
    name: 4.8-x86_64-priv
    reason: latest RHCOS build is 48.84.202109161436-0 which differs from istag 48.84.202107301701-0
-   changed: false
    name: 4.8-ppc64le
    reason: latest RHCOS build is still 48.84.202109161620-0 -- no change from istag
-   changed: true
    name: 4.8-ppc64le-priv
    reason: latest RHCOS build is 48.84.202109161620-0 which differs from istag 48.84.202107302004-0
-   changed: false
    name: 4.8-s390x
    reason: latest RHCOS build is still 48.84.202109161304-0 -- no change from istag
-   changed: true
    name: 4.8-s390x-priv
    reason: latest RHCOS build is 48.84.202109161304-0 which differs from istag 48.84.202107301721-0
```

is build-sync not updating the `-priv` imagestreams? i believe this would flag RHCOS as perpetually requiring an update.

**Comment by sosiouxme** [comment]
/hold cancel
build-sync indeed was not syncing `-priv` streams. that got fixed, we're ready for this. (would be nice to have reviews)

File: doozerlib/cli/scan_sources.py
Comment (jupierce) [inline doozerlib/cli/scan_sources.py:309]: Seems like we could narrow this catch. It hid one bug and could hide another.

**Review by jupierce — APPROVED** [review]
One thought, but otherwise lgtm.

File: doozerlib/cli/scan_sources.py
Comment (sosiouxme) [reply doozerlib/cli/scan_sources.py:309]: tough to narrow, there's a lot that can go wrong during the attempt. i guess we could start with obvious and add more as we encounter them.

i also would like to get around to unit testing this specifically...

**Comment by sosiouxme** [comment]
i pushed the original commit directly to master, but the followup commits improve the error handling and unit testing so we're less likely to see this again.

**Pull request merged**
